Hippo.Web/ClientApp/src/components/Admin/Sponsors.tsx

```diff
@@ -53,38 +53,52 @@ export const Sponsors = () => {
       body: JSON.stringify(request),
     });
 
-    setNotification(req, "Saving", "Sponsor Added/Updated");
-
-    const response = await req;
-
-    if (response.ok) {
-      const newAccount = await response.json();
-      let updatedAccounts = [];
-      //check if the newAccount is already in the list
-      if (accounts?.find((a) => a.id === newAccount.id)) {
-        //if it is, update the account
-        updatedAccounts = accounts
-          ? accounts.map((a) => (a.id === newAccount.id ? newAccount : a))
-          : [newAccount];
-      } else {
-        //if it is not, add it to the list and sort it
-        updatedAccounts = accounts ? [...accounts, newAccount] : [newAccount];
+    setNotification(
+      req,
+      "Saving",
+      async () => {
+        const response = await req;
+        let message = "";
+        if (response.ok) {
+          const newAccount = await response.json();
+          let updatedAccounts = [];
+
+          //check if the newAccount is already in the list
+          if (accounts?.find((a) => a.id === newAccount.id)) {
+            //if it is, update the account
+            updatedAccounts = accounts
+              ? accounts.map((a) => (a.id === newAccount.id ? newAccount : a))
+              : [newAccount];
+            message = "Sponsor Updated";
+          } else {
+            //if it is not, add it to the list and sort it
+            updatedAccounts = accounts
+              ? [...accounts, newAccount]
+              : [newAccount];
+            message = "Sponsor Added";
+          }
+          //sort the list
+          setAccounts(
+            updatedAccounts.sort((a, b) => a.name.localeCompare(b.name))
+          );
+
+          setRequest((r) => ({ ...r, lookup: "" }));
+          setRequest((r) => ({ ...r, name: "" }));
```

Comment:
don't do two in a row, do them both together (same reason as before).  `setRequest(r=>({ ...r, lookup: "", name: ""}));`